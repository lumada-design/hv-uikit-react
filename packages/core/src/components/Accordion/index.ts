export { default as accordionClasses } from "./accordionClasses";
export * from "./accordionClasses";
export * from "./Accordion";
