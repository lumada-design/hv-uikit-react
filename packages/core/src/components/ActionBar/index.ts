export { default as actionBarClasses } from "./actionBarClasses";
export * from "./actionBarClasses";
export * from "./ActionBar";
