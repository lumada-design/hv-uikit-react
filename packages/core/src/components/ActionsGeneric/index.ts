export { default as actionsGenericClasses } from "./actionsGenericClasses";
export * from "./actionsGenericClasses";
export * from "./ActionsGeneric";
