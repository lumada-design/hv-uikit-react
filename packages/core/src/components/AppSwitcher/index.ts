export * from "./AppSwitcher";
export { HvAction as HvAppSwitcherAction } from "./Action";
