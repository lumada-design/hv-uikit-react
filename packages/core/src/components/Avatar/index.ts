export { default as avatarClasses } from "./avatarClasses";
export * from "./avatarClasses";
export * from "./Avatar";
