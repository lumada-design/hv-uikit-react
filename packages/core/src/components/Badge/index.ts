export { default as badgeClasses } from "./badgeClasses";
export * from "./badgeClasses";
export * from "./Badge";
