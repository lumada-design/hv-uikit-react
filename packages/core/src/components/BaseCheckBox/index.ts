import { getClasses } from "utils";

export type HvBaseCheckBoxClasses = {
  root?: string;
  disabled?: string;
  focusVisible?: string;
  icon?: string;
};

const classKeys: string[] = ["root", "disabled", "focusVisible", "icon"];

export const baseCheckBoxClasses = getClasses<HvBaseCheckBoxClasses>(
  classKeys,
  "HvBaseCheckBox"
);

export * from "./BaseCheckBox";
