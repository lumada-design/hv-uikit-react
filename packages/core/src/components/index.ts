export * from "./Box";
export { Box as HvBox } from "./Box";
export * from "./Button";
export { Button as HvButton } from "./Button";
export * from "./Dropdown";
export { Dropdown as HvDropdown } from "./Dropdown";
export * from "./Header";
export {
  Actions as HvHeaderActions,
  Brand as HvHeaderBrand,
  Header as HvHeader,
  Navigation as HvHeaderNavigation,
} from "./Header";
export * from "./Typography";
export { Typography as HvTypography } from "./Typography";
