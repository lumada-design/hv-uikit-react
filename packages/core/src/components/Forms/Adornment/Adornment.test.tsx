import { CloseXS } from "@hitachivantara/uikit-react-icons";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it, vi } from "vitest";

import { HvAdornment } from "@core/components";

describe("Adornment", () => {
  it("should render the passed icon", () => {
    render(
      <HvAdornment icon={<CloseXS role="img" aria-label="close icon" />} />
    );
    expect(screen.getByRole("img", { name: "close icon" })).toBeInTheDocument();
  });

  it("should render a button if a 'onClick' is passed", async () => {
    const mockFn = vi.fn();
    render(<HvAdornment icon={<CloseXS />} onClick={mockFn} />);

    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();

    await userEvent.click(button);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
