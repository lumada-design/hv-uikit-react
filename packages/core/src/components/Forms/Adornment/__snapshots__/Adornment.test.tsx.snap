// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Adornment > should render correctly 1`] = `
<div>
  <div
    class="HvAdornment-root HvAdornment-adornment HvAdornment-adornmentIcon css-16q1xu7-StyledRoot ek2tbgi2"
    role="presentation"
  >
    <div
      class="HvAdornment-icon css-1431vi8-StyledIcon ek2tbgi1"
    >
      <div
        class="css-f95wjz"
        name="CloseXS"
      >
        <svg
          focusable="false"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="color0"
            d="m10.6 2.1-.7-.7L6 5.3 2.1 1.4l-.7.7L5.3 6 1.4 9.9l.7.7L6 6.7l3.9 3.9.7-.7L6.7 6z"
            fill="var(--uikit-colors-acce1)"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
